core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3801,6 +3801,126 @@ private ExpressionNode rewriteAggregate(ExpressionNode agg, QueryModel model) {
         return agg;
     }
 
+    /**
+     * Rewrites
+     * SELECT LAST(timestamp) FROM table_name;
+     * AND
+     * SELECT max(timestamp) FROM table_name;
+     * into query which can is optimised search
+     * SELECT timestamp from (SELECT timestamp from table_name order by timestamp) limit 1
+     * This enables FrameBackwardScan for aggregation queries containing LAST keyword
+     * AND
+     * Rewrites
+     * SELECT FIRST(timestamp) FROM table_name;
+     * AND
+     * SELECT min(timestamp) FROM table_name;
+     * into query which can is optimised search
+     * SELECT timestamp from (SELECT timestamp from table_name) limit 1
+     * This disables invoking group by workers
+     */
+    private QueryModel rewriteGroupByForFirstLastMaxMinAggregateFunctions(QueryModel parent) {
+        //base condition to stop recursion
+        if (parent == null)
+            return parent;
+
+        QueryModel modifiedNestedModel = parent.getNestedModel();
+        if (parent.getNestedModel() != null) {
```

Comment:
The condition should be more complicated since we target "simple" queries/sub-queries, i.e. queries operating on a "physical" table. Something like:
```java
    nestedModel != null
      && nestedModel.getJoinModels().size() == 1
      && nestedModel.getUnionModel() == null
      && nestedModel.getNestedModel() == null
      && nestedModel.getTableName() != null
      && nestedModel.getSampleBy() == null
      && nestedModel.getGroupBy().size() == 0
```